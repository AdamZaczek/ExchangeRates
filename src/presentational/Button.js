import React from 'react';

const Button = props => (
  <button onClick={props.onClick}>{props.children}</button>
);

export default Button;
