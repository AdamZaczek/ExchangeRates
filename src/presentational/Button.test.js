import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

test('renders correctly without any props', () => {
  const tree = renderer.create(
    <Button>ClickMe</Button>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with props', () => {
  const tree = renderer.create(
    <Button onClick={() => console.log('yay')}>ClickMe</Button>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});