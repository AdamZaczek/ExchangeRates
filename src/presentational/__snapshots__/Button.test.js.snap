// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with props 1`] = `
<button
  onClick={[Function]}
>
  ClickMe
</button>
`;

exports[`renders correctly without any props 1`] = `
<button
  onClick={undefined}
>
  ClickMe
</button>
`;
